qiskit/transpiler/preset_passmanagers/level1.py

```diff
@@ -105,14 +109,63 @@ def level_1_pass_manager(pass_manager_config: PassManagerConfig) -> PassManager:
     # 1. Use trivial layout if no layout given
     _given_layout = SetLayout(initial_layout)
 
-    _choose_layout_and_score = [
-        TrivialLayout(coupling_map),
-        Layout2qDistance(coupling_map, property_name="trivial_layout_score"),
-    ]
-
     def _choose_layout_condition(property_set):
         return not property_set["layout"]
 
+    def _trivial_not_perfect(property_set):
+        # Verify that a trivial layout is perfect. If trivial_layout_score > 0
+        # the layout is not perfect. The layout is unconditionally set by trivial
+        # layout so we need to clear it before contuing.
+        if property_set["trivial_layout_score"] is not None:
+            if property_set["trivial_layout_score"] != 0:
+                return True
+            else:
+                warnings.warn(
+                    "The current implicit default of using a trivial layout if it's a "
+                    "perfect layout will change in a future release, if you're "
+                    "depending on this behavior it is better to explicit set "
+                    "layout_method='trivial' when calling transpile()",
+                    FutureWarning,
+                )
+                # stack_level is not set because targeting transpile() is difficult
+        return False
```

Comment:
What's the intended path if the user doesn't depend on this behaviour, but wants to suppress the warning?

Reply:
I was assuming something like `warnings.filterwarnings("ignore", category="FutureWarning", module="qiskit.transpiler.preset_passmanagers.level1")` but I didn't really think about it too much. Were you thinking of adding a flag, env variable, or something else to either skip the initial layout check or suppress the warning?

Reply:
The differentiating factor to consider here (vs just changing the default `layout_method` value), is using a trivial layout first as an implicit default (if it's a perfect mapping) is a very observable thing, and has implications for creating physical circuits by hand. Then as I found with the tutorials, the documentation, and the tests a lot (but not all) of the examples and testing for physical circuits (especially where scheduling or sequencing) is involved just implicitly assume that calling a bare `transpile()` will return a trivial layout if it matches the target device constraints. This makes it tricky around how we document the change here. I agree for most users they won't care, the quality of the output just gets better for some circuits and a warning is just a big irritation for them (especially since it's not something straightforward to suppress). But for people who are depending on this behavior (especially as it's all over the documentation) it's critical that we warn that their code will break when we remove this.

With a change in the default value for `layout_method` it's a bit different because those algorithms are all heuristic and have different tradeoffs in quality vs runtime (most are stochastic too and behave randomly unless you fix a seed). Changing the algorithm is much less disruptive because unless you're depending on the exact transpilation being 100% reproducible between release you probably won't notice. They're also trivial to revert to the previous behavior with just setting the `layout_method` flag to previous default.

That being said typing all of this I might have convinced myself that maybe changing the behavior here just isn't the right choice. VF2Layout is noise aware and could provide better results than a trivial perfect layout but for optimization level 1 that tradeoff might be worth while. I still contend we shouldn't bother with it at higher optimization levels, but maybe for level 1 it's an ok tradeoff. @kdk @ajavadia do you have any thoughts here?

Reply:
Agree that the current contract to users is very fragile here, but is also deeply seated and heavily relied upon so I'd want to think a bit more before we'd think about how to change it.

If I understand correctly, the benefit to removing the `try_trivial_layout` behavior would be for the case where a trivial layout pre-routes a circuit, but so does some other layout with favorable noise characteristics that `VF2Layout` could find. So a user might unknowingly miss out on that optimization (though likely still be better off than if they had fallen back to one of the other layout passes).

From my perspective, the weight of adding something like `try_trivial_layout` to manage the rollout of this change seems to outweigh the missed optimization above, especially since it looks like this would ask nearly everyone to set `try_trivial_layout=False` across all their calls to `transpile`, so I'd prefer to keep this behavior as-is for now. Can you open an issue so we can track and discuss if and when we'd like to make this change?

Reply:
Yeah that sums it up pretty well. I think we're in agreement then to remove the warning and leave the level 1 behavior as is for now and I'll open a tracking issue after this PR merges to figure out how we migrate away from implicitly using `TrivialLayout` if it pre-routes the circuit in the future.